### PR TITLE
Add Grounding Skill to Samples

### PR DESCRIPTION
**Pull request by riedgar-ms**
### Motivation and Context

This introduces functions related to groundedness checking, based on learnings from the Aether Central team. The skills are showcased in both an Python notebook and a .NET sample.

### Contribution Checklist

- [x] The code builds clean without any errors or warnings
- [x] The PR follows SK Contribution Guidelines (https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md)
- [x] The code follows the .NET coding conventions (https://learn.microsoft.com/dotnet/csharp/fundamentals/coding-style/coding-conventions) verified with `dotnet format`
- [x] All unit tests pass, and I have added new tests where possible
- [x] I didn't break anyone :smile:

### CODE REVIEW

**Comment by riedgar-ms** [comment]
Thank you @shawncal . As we discussed, there are actually two ways of doing the groundedness checking in this PR. One has some simple single input/output skills in place, and usually works with the planner.

However, as we discussed, the single input/output constraint (with everything else in the context needing to be known at planning time) was giving trouble for a more sophisticated scenario. So, I created the second approach, which smuggles multiple arguments in and out of each function call. This obviously comes at the cost of much reduced flexibility.

@lemillermicrosoft  and @shawncal , I'd be happy to update this to match what you feel would best conform to your needs.

File: python/semantic_kernel/planning/basic_planner.py [hunk 1]
```diff
@@ -194,6 +194,9 @@ async def execute_plan_async(self, plan: Plan, kernel: Kernel) -> str:
         subtasks = generated_plan["subtasks"]
 
         for subtask in subtasks:
+            if debug_print:
```
Comment (riedgar-ms) [inline python/semantic_kernel/planning/basic_planner.py:197]: I've found these useful, to watch what's happening through a plan's execution, but I'm happy to remove these changes from the PR.

**Review by lemillermicrosoft — CHANGES_REQUESTED** [review]
I've suggested some changes to the skills and added some dotnet examples using skill individually and with a planner.

https://github.com/riedgar-ms/semantic-kernel/pull/1

**Comment by lemillermicrosoft** [comment]
> I've suggested some changes to the skills and added some dotnet examples using skill individually and with a planner.
> 
> [riedgar-ms#1](https://github.com/riedgar-ms/semantic-kernel/pull/1)

cc @alexchaomander and @awharrison-28 -- see discussion in the linked PR for more context -- tldr -- parity with planner in dotnet and python might be good to increase priority on https://github.com/orgs/microsoft/projects/865/views/3?pane=issue&itemId=30414768

**Review by lemillermicrosoft — DISMISSED** [review]

**Review by lemillermicrosoft — DISMISSED** [review]

File: samples/notebooks/python/09-groundedness-checking.ipynb [hunk 2]
```diff
@@ -0,0 +1,351 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "f5c76c5f",
+   "metadata": {},
+   "source": [
+    "# Groundedness Checking Skills\n",
+    "\n",
```
Comment (alexchaomander) [inline samples/notebooks/python/09-groundedness-checking.ipynb:9]: Can you write a bit more inside the notebook about what the concept of groundedness is and why it's important? I don't think everyone will know from the get-go about the subject.

Comment (riedgar-ms) [reply samples/notebooks/python/09-groundedness-checking.ipynb:9]: Good point - done.

**Review by alexchaomander — APPROVED** [review]
Looks good to me! Thank you for adding this important skill to SK!

**Pull request merged**